frontend/app/src/components/dashboard/EditBalancesSnapshotForm.vue

```diff
@@ -158,26 +160,51 @@ watch(assetType, assetType => {
   }
 });
 
-const categoryRules = [
-  (v: string) =>
-    !!v ||
-    t('dashboard.snapshot.edit.dialog.balances.rules.category').toString()
-];
-const assetRules = [
-  (v: string) =>
-    !!v || t('dashboard.snapshot.edit.dialog.balances.rules.asset').toString(),
-  (v: string) =>
-    !get(excludedAssets).includes(v) ||
-    t(
-      'dashboard.snapshot.edit.dialog.balances.rules.asset_non_duplicate'
-    ).toString()
-];
-const amountRules = [
-  (v: string) =>
-    !!v || t('dashboard.snapshot.edit.dialog.balances.rules.amount').toString()
-];
-const valueRules = [
-  (v: string) =>
-    !!v || t('dashboard.snapshot.edit.dialog.balances.rules.value').toString()
-];
+const rules = {
+  category: {
+    required: helpers.withMessage(
+      t('dashboard.snapshot.edit.dialog.balances.rules.category').toString(),
+      required
+    )
+  },
+  asset: {
+    required: helpers.withMessage(
+      t('dashboard.snapshot.edit.dialog.balances.rules.asset').toString(),
+      required
+    ),
+    nonDuplicate: helpers.withMessage(
+      t(
+        'dashboard.snapshot.edit.dialog.balances.rules.asset_non_duplicate'
+      ).toString(),
+      (v: string): boolean => !get(excludedAssets).includes(v)
+    )
+  },
+  amount: {
+    required: helpers.withMessage(
+      t('dashboard.snapshot.edit.dialog.balances.rules.amount').toString(),
+      required
+    )
+  },
+  value: {
+    required: helpers.withMessage(
+      t('dashboard.snapshot.edit.dialog.balances.rules.value').toString(),
+      required
+    )
+  }
+};
+
+const v$ = useVuelidate(
+  rules,
+  {
+    category: computed(() => get(form).category),
+    asset: computed(() => get(form).assetIdentifier),
+    amount: computed(() => get(form).amount),
+    value: computed(() => get(form).usdValue)
```

Comment:
mostly out of curiosity, do you think this would work with something like https://vueuse.org/shared/reactivepick/